exposed-core/src/main/kotlin/org/jetbrains/exposed/dao/id/IdTable.kt

```diff
@@ -37,19 +37,49 @@ abstract class IdTable<T : Comparable<T>>(name: String = "") : Table(name) {
 
     /** All base columns that make up this [IdTable]'s identifier column. */
     val idColumns = HashSet<Column<out Comparable<*>>>()
+        get() = field.validate()
```

Comment:
If anybody else has thoughts about the mutability of `idColumns`, I'm open to alternatives.

Reply:
All `IdTable` are available within the core module. So users can access them, for convenience, even if they don't use DAO. So a user that already has been using composite primary keys in DSL could choose now to convert it to a `CompositeIdTable` to take advantage of the `id` column. So there needs to be a way to do that. If `entityid()` is moved to DAO, how will it initialize all the `IdTable` with final properties?

Reply:
I got the point, `CompositeIdTable` as an better alternative to simple `Table` with multiple columns in primary key sounds good and similar to other `IdTables`.